File: Module/src/main/java/com/example/william/my/module/bean/LoginData.java
package com.example.william.my.module.bean;

import androidx.annotation.Keep;

import com.google.gson.annotations.SerializedName;

@Keep
public class LoginData {

    @SerializedName("id")
    private String id;
    @SerializedName("nickname")
    private String nickname;

    public String getId() {
        return id;
    }

    public String getNickname() {
        return nickname == null ? "" : nickname;
    }
}

File: Modules/Kotlin/src/main/kotlin/com/example/william/my/module/kotlin/bean/LoginData.kt
package com.example.william.my.module.kotlin.bean

import android.os.Parcelable
import androidx.annotation.Keep
import kotlinx.parcelize.Parcelize

@Keep
@Parcelize
class LoginData(val data: User) : Parcelable {

    @Keep
    @Parcelize
    class User(val id: String, val nickname: String) : Parcelable
}